### ISSUE TITLE
hyperparameter tuning 

### ISSUE THREAD

**Issue by saeed48104617** [AI4Finance-LLC/Deep-Reinforcement-Learning-for-Automated-Stock-Trading-Ensemble-Strategy-ICAIF-2020#10]
hey sir thanks for your helpful documentation 
 I have a question and i was wondering if yo could answer 
have you tried hyperparameter tuning like hyperparameter tuning  and why you havent used this in your project 

thanks

**Comment by BruceYanghy** [comment]
> hey sir thanks for your helpful documentation
> I have a question and i was wondering if yo could answer
> have you tried hyperparameter tuning like bayesian optimization and why you havent used this in your project
> 
> thanks

I tried Grid Search and Random Search. What I didn't do in this paper is: tuning the parameters along the way, I only used the in-sample data from 2009-2015 to tune the parameters once, so there is some alpha decay here. Ideally, you should tune the parameters every quarter when you retrain your model.

Numerous hyperparameters – e.g. the learning rate, total_timesteps – influence the learning process and are usually determined by testing some variations.
